### PR TITLE
DM-46049: Add a Slack webhook to notify on failure

### PR DESCRIPTION
**Pull request by arunkannawadi**

### CODE REVIEW

File: .github/workflows/lsst-tests.yml [hunk 1]
```diff
@@ -60,3 +59,19 @@ jobs:
         uses: codecov/codecov-action@v3
```
Comment (timj) [inline .github/workflows/lsst-tests.yml:59]: You are going to want to update your codecov action to v4 and include the per-repo token (see daf_butler).

File: .github/workflows/lsst-tests.yml [hunk 2]
```diff
@@ -60,3 +59,19 @@ jobs:
...
           files: ./coverage.xml
+          fail_ci_if_error: false
+
+  notify-on-failure:
+    runs-on: ubuntu-latest
+    if: ${{ always() && (github.event_name == 'push' || github.event_name == 'schedule') && needs.lsst-tests.result == 'failure' }}
+    needs:
+      - lsst-tests
+    steps:
+    - uses: actions/checkout@v2
```
Comment (timj) [inline .github/workflows/lsst-tests.yml:70]:
```suggestion
    - uses: actions/checkout@v4
```

File: .github/workflows/lsst-tests.yml [hunk 3]
```diff
@@ -60,3 +59,19 @@ jobs:
...
+    - name: Slack Notification
```
Comment (timj) [inline .github/workflows/lsst-tests.yml:71]: I've never used this before so how does it know which channel to write to?

**Review by timj — APPROVED** [review]
I think you need to update your codecov action.

I've never used the Slack webhook but I assume it works.  I'm ignoring all the other unrelated changes.

File: .github/workflows/lsst-tests.yml (hunk 2 above)
Comment (arunkannawadi) [reply .github/workflows/lsst-tests.yml:70]: Ah, this is from upstream but good to change it here.

File: .github/workflows/lsst-tests.yml (hunk 3 above)
Comment (arunkannawadi) [reply .github/workflows/lsst-tests.yml:71]: I can add the app to any channel, or from the App management page I can add it to any channel.

**Comment by arunkannawadi** [comment]
Ignoring all the Python changes is fine. They are all from upstream and reviewed by Erin & Matt.

**Pull request merged**